discovery_child_development/pipeline/openalex/00a_openalex_metaflow_random.py

```diff
@@ -0,0 +1,143 @@
+"""
+Works pipeline
+--------------
+
+A pipeline that takes a list of years, and outputs a random sample of OpenAlex API results. 
+You can specify the number of works you want to retrieve per year, with a maximum of 10,000.
+
+Usage:
+
+First, amend these variables:
+* YEARS: list of years you want to retrieve publications from
+
+To test the flow with just the first year in the list:
+python discovery_child_development/pipeline/openalex/00_openalex_metaflow.py run --production False 
+
+To fetch the full dataset:
+python discovery_child_development/pipeline/openalex/00_openalex_metaflow.py run --production True 
+"""
+import requests
+from metaflow import FlowSpec, S3, step, Parameter, retry, batch
+from nesta_ds_utils.loading_saving import S3 as nesta_s3
+from dotenv import load_dotenv
+from typing import NoReturn, List, Any
+import random
+from discovery_child_development import PROJECT_DIR, S3_BUCKET, config
+
+API_ROOT = "https://api.openalex.org/works?filter="
+S3_PATH = "metaflow"
+load_dotenv()
+
+
+def generate_random_queries(years: List[str], number_works: int) -> List[str]:
+    """Generates a list of queries for a random list of works for the
+    years required.
+
+    Args:
+        years : list of years to be queried
+
+    Returns:
+        query_list : list of all queries
+    """
+    return [f"{year}&sample={number_works}" for year in years]
+
+
+def api_generator_random(api_root: str, filter_call: str) -> list:
+    """Generates a list of all URLs needed to collect a random sample
+
+    Args:
+        api_root : root URL of the OpenAlex API
+
+        concept_ids : list of concept IDs to be queried
```

Comment:
This can be removed I think

Reply:
Yep, this is what I get for quickly adding the file in haha!